### DISCUSSION TITLE
How to quick share weight like the tensorflow using reused_params !!!

### DISCUSSION THREAD

**Discussion by Yezipiaomu** [apache/incubator-mxnet#8606] [Q&A]
I want to share a vgg params with two different inputs! but I find it's not easy to quickly share weight !!
Anyone can tell me a quick solution like the tensorflow using reusing the weight!!!

**Comment by jiarenyf** [comment]
I also want to share params with two different inputs, but I find it's not easy to quickly share weight ?
Anyone can tell me a quick solution like the tensorflow using reusing the weight ? Thanks !

**Comment by saicoco** [comment]
Maybe you can refer to the implementation of GAN

**Comment by szha** [comment]
@apache/mxnet-committers: This issue has been inactive for the past 90 days. It has no label and needs triage.

For general "how-to" questions, our [user forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) is a good place to get help.

**Comment by thomelane** [comment]
With Gluon you can just call the same `Block` multiple times on different inputs. And `autograd` will manage the gradient calculations so the the parameters of the Block can be updated correctly, even when the `Block` is called multiple times. So taking VGG16 as an example, and applying it to two different inputs and concatenating the results:

```
import mxnet as mx

# creating dummy data
data1 = mx.nd.random.normal(shape=(10,3,32,32))
data2 = mx.nd.random.normal(shape=(10,3,32,32))

# creating (and initializing) the block
vgg_model = mx.gluon.model_zoo.vision.vgg16()
vgg_model.collect_params().initialize(mx.init.Xavier())

# calling the block twice (and combining)
out1 = vgg_model(data1)
out2 = vgg_model(data2)
out = mx.nd.concatenate([out1, out2], axis=1)
```

Just perform the last steps inside an `autograd.record()` context and you'll be able to update the parameters of the VGG16 network jointly. One thing to note is due to the fully connected layers at the end of the VGG16 model, the shape of the inputs must match (i.e. both were 32x32 in this example). Other models wouldn't have this limitation if they use adaptive pooling methods.
